### ISSUE TITLE
Cannot sign into new installation

### ISSUE THREAD

**Issue by gljones2001** [papermerge/papermerge-core#17]
I have installed the Portainer setup method, with docker. 

The issue I am currently having though is this:
![image](https://user-images.githubusercontent.com/20365116/165557982-cff929a6-4743-4b59-84ed-dfa03b5378ce.png)

I cannot get any further. It locks me out for some reason. Here are the logs:
[_papermerge_redis_1_logs.txt](https://github.com/papermerge/papermerge-core/files/8573529/_papermerge_redis_1_logs.txt)
[_papermerge_db_1_logs (1).txt](https://github.com/papermerge/papermerge-core/files/8573530/_papermerge_db_1_logs.1.txt)
[_papermerge_es_1_logs (1).txt](https://github.com/papermerge/papermerge-core/files/8573531/_papermerge_es_1_logs.1.txt)
[_papermerge_frontend_1_logs.txt](https://github.com/papermerge/papermerge-core/files/8573532/_papermerge_frontend_1_logs.txt)
[_papermerge_ws_server_1_logs.txt](https://github.com/papermerge/papermerge-core/files/8573533/_papermerge_ws_server_1_logs.txt)
[_papermerge_backend_1_logs.txt](https://github.com/papermerge/papermerge-core/files/8573534/_papermerge_backend_1_logs.txt)
[_papermerge_worker_1_logs.txt](https://github.com/papermerge/papermerge-core/files/8573535/_papermerge_worker_1_logs.txt)
[_papermerge_backend_init_1_logs (1).txt](https://github.com/papermerge/papermerge-core/files/8573536/_papermerge_backend_init_1_logs.1.txt)
[_portainer_logs.txt](https://github.com/papermerge/papermerge-core/files/8573537/_portainer_logs.txt)
[_papermerge_traefik_1_logs.txt](https://github.com/papermerge/papermerge-core/files/8573538/_papermerge_traefik_1_logs.txt)

Please help. I don't know what to do at this point.

**Comment by gljones2001** [comment]
Hmm, yeah... I am using Ubuntu server 20.04 LTS with docker and Portainer installed. It has no GUI (being server) so I have to use a separate PC to connect to it. I did not change any of the config (Besides the other issue you mentioned), so I don't know what could be causing it.

**Comment by gljones2001** [comment]
> 

Ohhhhh I just looked at which log you were talking about. That error is because I am using portainer locally only, and it is using a self-generated TLS certificate. I have to click the "proceed to this site anyways" in chrome when I go to the portainer web portal.

**Comment by gljones2001** [comment]
Ok, the issue still happens. I used the new setup, with the `HOSTNAME` variable, and it now has the old `404: File Not Found` error again.

**Comment by gljones2001** [comment]
Would this have anything to do with the error?

```

2022-05-03 13:32:39.193 UTC [69] STATEMENT:  SELECT "core_user"."password", "core_user"."last_login", "core_user"."is_superuser", "core_user"."username", "core_user"."first_name", "core_user"."last_name", "core_user"."email", "core_user"."is_staff", "core_user"."is_active", "core_user"."date_joined", "core_user"."id", "core_user"."role_id", "core_user"."home_folder_id", "core_user"."inbox_folder_id", "core_user"."created_at", "core_user"."updated_at" FROM "core_user" WHERE "core_user"."username" = 'admin' LIMIT 21,
2022-05-03 13:32:39.193 UTC [69] ERROR:  relation "core_user" does not exist at character 416
```

**Comment by gljones2001** [comment]
Also noticed a bunch of python errors in the `backend_init` container:
[_papermerge_backend_init_1_logs (2).txt](https://github.com/papermerge/papermerge-core/files/8651769/_papermerge_backend_init_1_logs.2.txt)
.

**Comment by gljones2001** [comment]
AHA! I found something! the 1st line in that `backent_init` log is:
`File "/venv/lib/python3.8/site-packages/django/db/backends/base/base.py", line 603, in temporary_connection, django.db.utils.ProgrammingError: relation "core_user" does not exist`
And the error in the Database is:
```
2022-05-03 13:32:39.193 UTC [69] STATEMENT:  SELECT "core_user"."password", "core_user"."last_login", "core_user"."is_superuser", "core_user"."username", "core_user"."first_name", "core_user"."last_name", "core_user"."email", "core_user"."is_staff", "core_user"."is_active", "core_user"."date_joined", "core_user"."id", "core_user"."role_id", "core_user"."home_folder_id", "core_user"."inbox_folder_id", "core_user"."created_at", "core_user"."updated_at" FROM "core_user" WHERE "core_user"."username" = 'admin' LIMIT 21,
2022-05-03 13:32:39.193 UTC [69] ERROR:  relation "core_user" does not exist at character 416
```
Notice the `core_user` errors correlate. There is an error creating the `core_user` field in the database, and then the backend fails to read it because `core_user` doesn't exist.

**Comment by ciur** [comment]
> Also noticed a bunch of python errors in the `backend_init` container: [_papermerge_backend_init_1_logs (2).txt](https://github.com/papermerge/papermerge-core/files/8651769/_papermerge_backend_init_1_logs.2.txt)

In log there is error message saying:
```
django.db.utils.OperationalError: could not connect to server: Connection refused

	Is the server running on host "db" (172.20.0.6) and accepting

	TCP/IP connections on port 5432?
```
Which means that for some reason it connect connect to database. In docker compose file, [backend_init depends on db](https://github.com/papermerge/papermerge-core/blob/master/docker/docker-compose.yml#L43).

> There is an error creating the core_user relation in the database, and then the backend fails to read it because core_user doesn't exist.

Because database was not up and running.

**Comment by gljones2001** [comment]
Huh. The database is running. I don't know why it would be connecting.

**Comment by gljones2001** [comment]
Have you looked into doing something like this: [https://docs.docker.com/compose/networking/#links](https://docs.docker.com/compose/networking/#links)

If you add a link to the `backend_init` compose section connecting it to `db`, that might help IDK.

**Comment by cupracer** [comment]
> Ok, the issue still happens. I used the new setup, with the `HOSTNAME` variable, and it now has the old `404: Page Not Found` error again.

I discovered Papermerge today and gave it a try. This setup was started via docker-compose on a dedicated SLES 15 SP3 host. I experienced the same issues as described here during startup ("core_user" missing, db connection refused... seem to be irrelevant here) and also that "404: Page Not Found" error.

I configured "HOSTNAME" in .env as a full-qualified hostname ("somehost.my.domain"). Using the FQDN in the browser lead to that 404 error while using just the short name worked perfectly and presented the login page as expected.
Searching for an explanation revealed that docker-compose just used the local part "somehost" when ${HOSTNAME} was used in docker-compose.yml. 

I finally solved this 404 error for my setup and it seems that #22 is broken by design (@ciur):

**"HOSTNAME" from the .env file is simply not used if that variable also exists in the shell environment** where docker-compose is executed. On my system "HOSTNAME" is set automatically and contains the local part of my system's hostname. 

[See docker-compose documentation regarding environment variables](https://docs.docker.com/compose/environment-variables/)

> You can set default values for environment variables using a [.env file](https://docs.docker.com/compose/env-file/), which Compose automatically looks for in project directory (parent folder of your Compose file). **Values set in the shell environment override those set in the .env file.**

I fixed this by just renaming the variable HOSTNAME to USE_HOSTNAME in .env and docker-compose.yml.
